API Absolute Paths

droghio opened the pull request:
Validates that file paths sent to the renter and wallet apis are absolute paths. Throws an error otherwise.

The host module has implemented this behavior since pr #1325. I am not sure if we should move this check to the api to or the wallet/host checks into the modules so in the mean time I have left the host api as is.

Fixes #1385.

mnsl commented:
Maybe include tests, e.g.
```go
func TestRenterRelativePathError(t *testing.T) {
	if testing.Short() {
		t.SkipNow()
	}
	st, err := createServerTester("TestRenterRelativePathError")
	if err != nil {
		t.Fatal(err)
	}
	defer st.server.Close()

	// Announce the host.
	if err := st.announceHost(); err != nil {
		t.Fatal(err)
	}

	// Create a file in the present working directory.
	if err = createRandFile("test.dat", 1024); err != nil {
		t.Fatal(err)
	}

	// Upload to host.
	uploadValues := url.Values{}
	uploadValues.Set("source", "test.dat")
	err = st.stdPostAPI("/renter/upload/foo/bar/test", uploadValues)
        if err == nil || err.Error() != "source must be an absolute path" {
		t.Fatal(err)
	}
        // etc....
}

If this seems like too much of a hassle, I'm happy to do the rest

DavidVorick commented:
Testing needs to accompany all PRs, but wouldn't be difficult to add a commit from @mnsl to this branch

lukechampine commented:
unfortunately this branch is on drogo's fork, so mnsl can't push to it directly

droghio commented:
Makes sense, @mnsl gave a great example of a test and I'll work on extending it to cover all modified API routes. In the mean time I also gave @mnsl push permissions to my repo, just in case.

droghio commented:
Added tests and rebased to latest master. The hosts module still implements an absolute path check in the module instead of the API. Not sure if this should be moved/duplicated in the api or left as is, but I feel where we perform these checks should be consistent.
